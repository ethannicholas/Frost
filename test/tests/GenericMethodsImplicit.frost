package test1.test2.test3

class GenericMethodsImplicit {
	@class
	method indexOf<T:Equatable<T>>(list:ListView<T>, needle:T):Int? {
		for i in 0 .. list.count {
			if list[i] = needle {
				return i
			}
		}
		return null
	}

	@class
	method main() {
		Console.printLine(indexOf([1, 7, 2, 9, -6, 0, 5, 8, 10], 5))
	    Console.printLine(Int[1 ... 5].combine(Int[1 ... 5], (a, b) => a + b))
	    Console.printLine(Int[1 ... 5].combine(Int[1 ... 5], Int.*)
	}
}