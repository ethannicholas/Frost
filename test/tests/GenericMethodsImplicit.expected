6
[2, 4, 6, 8, 10]
[1, 4, 9, 16, 25]
